### DISCUSSION TITLE
Mixed vehicles

### DISCUSSION THREAD

**Discussion by TeodoraBogdan** [google/or-tools#3588] [Routing (and legacy CP) questions]
Hello!

Can someone help me with a solution to this problem?

For example I have 100 stops (some are inside of a city and some are outside) and 2 vehicles, one car and one truck. Because truck does not have access to the same roads as a car I have 2 different distance matrices.
I want to optimize this problem, so the stops outside of the city to be assigned to the truck and the ones inside the city to the car.

Please tell me any idea you might have...I really need to make this work. Maybe there is a workaround...

**Comment by jmarca** [comment]
I’m not sure I understand your question. Are you asking if vehicles can have different cost functions?  If so, yes. Read the API docs here: https://developers.google.com/optimization/reference/python/constraint_solver/pywrapcp#adddimensionwithvehicletransitandcapacity

This call lets you pass in a vector of callbacks not just one when creating a dimension. 

And then this API call 
https://developers.google.com/optimization/reference/python/constraint_solver/pywrapcp#setarccostevaluatorofvehicle

will allow you to define per-vehicle arc cost evaluators.

**Reply by Mizux** [reply]
On top of what jmarca wrote (many thx ;)) you can also use the following variable to limit the visit of a node to a subset of vehicle
https://github.com/google/or-tools/blob/b882a5f8732ec208e07769f57d5ac0f6236c9134/ortools/constraint_solver/routing.h#L1495-L1497

e.g. "outside of city nodes can only be visited by truck" etc...

note: you can also be interested in this samples:
[vrp_multiple_transit.py](https://gist.github.com/Mizux/6d9283388150020df7542adf63577786)
